Add configuration flag for  #141

ingwarr opened the pull request:
added enable_binary_authorization flag

alexkonkin commented:
> 
> 
> added enable_binary_authorization flag

Your lint tests failed because you should execute 2 commands before commit:
make generate
make generate_docs

alexkonkin commented:
Please also don't forget to add the tests that validate that your feature is present in the newly created cluster.

ingwarr commented:
@alexkonkin Thank you!

ingwarr commented:
> > added enable_binary_authorization flag
> 
> Your lint tests failed because you should have executed 2 commands before commit:
> make generate
> make generate_docs
Fixed, I've added appropriate variables to autogen

aaron-lane commented on autogen/cluster_regional.tf:
```diff
@@ -36,6 +36,9 @@ resource "google_container_cluster" "primary" {
   logging_service    = "${var.logging_service}"
   monitoring_service = "${var.monitoring_service}"
 
+{% if private_cluster %}
```
What is the motivation to only provide this for private clusters?

ingwarr replied:
Because of this change valid only for google-beta provider

aaron-lane replied:
Got it. Thanks!

aaron-lane commented on autogen/outputs.tf:
```diff
@@ -78,6 +78,13 @@ output "ca_certificate" {
+{% if private_cluster %}
+output "enable_binary_authorization" {
+description = "Enable BinAuthZ Admission controller"
+value = "${var.enable_binary_authorization}"
```
Generally, we do not need to output variables unless there are no other meaningful outputs of a module. This should be dropped.

ingwarr replied:
Ok, I'll do it

ingwarr replied:
Done

aaron-lane commented on autogen/variables.tf:
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
+{% if private_cluster %}
+variable "enable_binary_authorization" {
+description = "Enable BinAuthZ Admission controller"
+default     = false
```
```suggestion
  default     = false
```

aaron-lane commented on autogen/variables.tf:
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
 
+{% if private_cluster %}
+variable "enable_binary_authorization" {
+description = "Enable BinAuthZ Admission controller"
```
```suggestion
  description = "Enable BinAuthZ Admission controller"
```

aaron-lane commented on examples/deploy_service/README.md:
```diff
@@ -16,6 +16,7 @@ It will:
 |------|-------------|:----:|:-----:|:-----:|
 | cluster\_name\_suffix | A suffix to append to the default cluster name | string | `""` | no |
 | compute\_engine\_service\_account | Service account to associate to the nodes in the cluster | string | n/a | yes |
+| enable\_binary\_authorization | Enable BinAuthZ Admission controller | string | `"false"` | no |
```
These tables should be edited only through `make generate_docs`, not manually; this variable has not been added to any of the examples.

```suggestion
```

aaron-lane commented on autogen/outputs.tf:
```diff
@@ -78,6 +78,7 @@ output "ca_certificate" {
   value       = "${local.cluster_ca_certificate}"
 }
 
+
```
```suggestion
```

aaron-lane commented on autogen/variables.tf:
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
+  default     = false
+}
+{% endif %}
+
```
```suggestion
```

aaron-lane commented on autogen/variables.tf:
```diff
@@ -86,6 +86,13 @@ variable "master_authorized_networks_config" {
+variable "enable_binary_authorization" {
+  description = "Enable BinAuthZ Admission controller"
+  default     = false
+}
```
```suggestion
}

```

aaron-lane commented on modules/private-cluster/outputs.tf:
```diff
@@ -78,6 +78,11 @@ output "ca_certificate" {
   value       = "${local.cluster_ca_certificate}"
 }
 
+output "enable_binary_authorization" {
```
Please remove this output; it doesn't provide any value.

ingwarr replied:
Ok, done

alexkonkin commented:
Quoting the results of the integration-tests execution:
a)
google_compute_subnetwork.main: Error waiting to create Subnetwork: Error waiting for Creating Subnetwork: Quota 'ROUTES' exceeded.  Limit: 250.0 globally.
b)
Error waiting for Deleting Network: The network resource '<...>/networks/cft-gke-test-5pwt' is already being used by 'projects/ci-kubernetes-engine/regions/us-east4/subnetworks/cft-gke-test-5pwt'

It seems that some resources have not been deleted after the previous CI execution....
As far as I remember such situation can be resolved by the one who has
the administrator's permission to do this.

aaron-lane commented:
I manually verified the node pool suite; the failure is unrelated to these changes.
